### PR TITLE
I/o refactor

### PR DESCRIPTION
**Pull request by kRichard32**
Refactoring the code so output is handled in outer layers. A state object is now passed towards the outer layers and the outer layers deal with turning the state object into user friendly output.

### CODE REVIEW

File: src/main/java/Interactors/IOController.java [hunk 1]
```diff
@@ -68,18 +68,166 @@ public void connectLogic(){
...
+                break;
+            case "TradeTree":
+                break;
+            case "AcceptTrade":
+                outputString.append("Trade success");
+                break;
+            case "DeclineTrade":
+                outputString.append("Trade failure");
+                break;
+        }
```
Comment (harshkpatel) [inline src/main/java/Interactors/IOController.java:219]: I like how you moved all the logic out of the State Entity into IO controller as it emphasizes clean architecture. One thing a little worrying is that IOController might be handling too much now, but this is an issue for another refactor where we create another interface that deals with these values.

**Review by harshkpatel — APPROVED** [review]
Overall the changes look good and it seems easier to change and branch out IO functionality and features, which is great. I do not have any huge changes that need to be made right away as I am assuming you thoroughly tested the code on your own as there are no tests. There is no big issue with the tests right now as the logic seems to have many edge cases that we can debug after this change is implemented. I am also assuming that JavaDoc will be added to a later PR, because some of these changes are a little harder to follow without documentation. Good use of Comments as a placeholder while the JavaDoc is not present, it helped me understand what each line was doing.

File: src/main/java/Interactors/IOController.java (hunk 1 above)
Comment (kRichard32) [reply src/main/java/Interactors/IOController.java:219]: Exactly, we might want to soon create another class which lies in the outside layers which deal with this handling of state values.

**Review by derrcodes — APPROVED** [review]
Refactoring the code was a good choice because the changes help the implementation adhere to clean architecture. I notice that there is still a lack of JavaDoc and tests so that's something we have to add later.

File: src/main/java/Interactors/IOController.java [hunk 2]
```diff
@@ -68,18 +68,166 @@ public void connectLogic(){
      * Helper function that takes state data and formulates the data into a string readable by the user
      */
     public void displayOptions(){
+    //    StringBuilder outputString = new StringBuilder();
+ //       outputString.append(currentState.getDescription());
+        this.output = stateToString(currentState);
+
+    }
+    public String stateToString(State state){
         StringBuilder outputString = new StringBuilder();
```
Comment (Leuxll) [inline src/main/java/Interactors/IOController.java:77]: Perhaps have some comments on each one, but not it should be good in that it adheres to clean architecture. Also, we will need to shorten this using helper classes and also some other refactoring.

**Review by Leuxll — APPROVED** [review]
